From: Reinhard Tartler <siretart@tauware.de>
Date: Wed, 29 Jul 2026 18:54:43 +0200
Subject: Disable flaky compaction test on slow architectures

The TestDelayedCompaction test requires compaction to complete within 1
second, but on slower 32-bit architectures (i386, s390x) this can fail
due to timing sensitivity. Disable it for Debian builds.

Forwarded: not-needed
---
 tsdb/compact_test.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tsdb/compact_test.go b/tsdb/compact_test.go
index 655cfa4..363619d 100644
--- a/tsdb/compact_test.go
+++ b/tsdb/compact_test.go
@@ -1928,6 +1928,7 @@ func TestCompactEmptyResultBlockWithTombstone(t *testing.T) {
 }
 
 func TestDelayedCompaction(t *testing.T) {
+	t.Skip("Disabled for Debian build: flaky on slow architectures due to timing sensitivity")
 	// The delay is chosen in such a way as to not slow down the tests, but also to make
 	// the effective compaction duration negligible compared to it, so that the duration comparisons make sense.
 	delay := 1000 * time.Millisecond
